src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class FunctionPointerMethodSymbol : MethodSymbol
+    {
+        private readonly ImmutableArray<FunctionPointerParameterSymbol> _parameters;
+        private DiagnosticInfo? _lazyUseSiteDiagnostic = CSDiagnosticInfo.EmptyErrorInfo;
```

Comment:
>private DiagnosticInfo? _lazyUseSiteDiagnostic = CSDiagnosticInfo.EmptyErrorInfo; [](start = 8, length = 81)

I do not think we should be caching this for FunctionPointerMethodSymbol. We are caching this information only for definitions that do not belong to the current compilation. We also do not calculate that for definitions in source because those get all the required diagnostics while we bind syntax. Arrays, constructed types and regular pointer types are good examples. The symbols referring to the FunctionPointerMethodSymbol will merge information across all types involved and cache it as appropriate.